lib/perl/Genome/Model/Tools/Bsmap/MethylationRatio.pm

```diff
@@ -0,0 +1,99 @@
+package Genome::Model::Tools::Bsmap::MethylationRatio;
+
+use strict;
+use warnings;
+use Cwd;
+use Genome;
+
+use Genome::Utility::Text qw(sanitize_string_for_filesystem);
+use File::Spec;
+use File::Basename qw();
+
+my ( undef, $dir ) = File::Basename::fileparse(__FILE__);
+my $METHRATIO_COMMAND = File::Spec->join( $dir, 'methylation_ratio.py' );
+
+class Genome::Model::Tools::Bsmap::MethylationRatio {
+	is  => 'Command',
+	has => [
+		bam_file => {
+			is       => 'Text',
+			doc      => 'An indexed bam file',
+			is_input => 1,
+		},
+		output_file => {
+			is => 'Text',
+			doc =>
+'File name for methyl counts (a .bgz extension will be added automatically)',
+			is_output     => 1,
+			is_input      => 1,
+			default_value => 'methylation',
+		},
+		threads => {
+			is => 'Text',
```

Comment:
Here you could also use `Number` instead of `Text`; again, mostly it's just metadata.